autogen/agentchat/conversable_agent.py

```diff
@@ -321,6 +322,75 @@ def reply_func(
         if ignore_async_in_sync_chat and inspect.iscoroutinefunction(reply_func):
             self._ignore_async_func_in_sync_chat_list.append(reply_func)
 
+    @staticmethod
+    def simple_nested_chat(chat_queue, recipient, messages, sender, config):
+        """A simple chat reply function.
+        This function initiate one or a sequence of chats between the "recipient" and the agents in the
+        chat_queue.
+
+        It extracts and returns a summary from the nested chat based on the "summary_method" in each chat in chat_queue.
+        """
+        last_msg = messages[-1].get("content", "")
+        chat_to_run = []
+        for i, c in enumerate(chat_queue):
+            current_c = c.copy()
+            message = current_c.get("message")
+            if callable(message):
+                message = message(recipient, messages, sender, config)
+            # If message is not provided in chat_queue, we by default use the last message from the original chat history as the first message in this nested chat (for the first chat in the chat queue).
+            # NOTE: This setting is prone to change.
+            if message is None and i == 0:
+                message = last_msg
+            # We only run chat that has a valid message. NOTE: This is prone to change dependin on applications.
+            if message:
+                current_c["message"] = message
+                chat_to_run.append(current_c)
+        if not chat_to_run:
+            return True, None
+        res = recipient.initiate_chats(chat_to_run)
+        return True, res[-1].summary
+
+    def register_nested_chats(
+        self,
+        trigger,
+        chat_queue,
+        reply_func_from_nested_chat="auto",
+        position: int = 2,
+        **kwargs,
+    ):
+        """Register a nested chat reply function.
+        Args:
+            trigger (Agent class, str, Agent instance, callable, or list): Ref to `register_reply` for details.
```

Comment:
Can we allow `None` for the `trigger` and make it an optional? In most cases this is not needed and `[Agent, None]` is sufficient. We don't need to expose this detail for users who don't know about register_reply.  Ideally, `chat_queue` should be the only required argument.

Reply:
"In most cases [Agent, None] is sufficient." I am not sure about this. The OptiGuide example is using different triggers. 

```
# writer is triggered only when receiving a message from the user
optiguide_commander.register_nested_chats("user", writer_chat_queue)
# safeguard is triggered only when receiving a message from the writer
optiguide_commander.register_nested_chats("writer", safeguard_chat_queue)
```
In the chess example, different trigger would be used as well.

@sonichi  what do you think?

Reply:
Right, I might be incorrect to say "in most cases". Though I am wondering for the default case whether `[Agent, None]` is good. It helps the default interface looks simpler.